code/datums/elements/lazystorage.dm

```diff
@@ -0,0 +1,71 @@
+/datum/element/lazystorage
+
+/datum/element/lazystorage/Attach(datum/target)
+	. = ..()
+	if(!istype(target, /obj/item/storage))
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_CONTAINS_STORAGE, .proc/on_check)
+	RegisterSignal(target, COMSIG_IS_STORAGE_LOCKED, .proc/check_locked)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_SHOW, .proc/signal_show_attempt)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_INSERT, .proc/signal_insertion_attempt)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_CAN_INSERT, .proc/signal_can_insert)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_TAKE_TYPE, .proc/signal_take_type)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_FILL_TYPE, .proc/signal_fill_type)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_SET_LOCKSTATE, .proc/set_locked)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_TAKE, .proc/signal_take_obj)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_QUICK_EMPTY, .proc/signal_quick_empty)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_HIDE_FROM, .proc/signal_hide_attempt)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_HIDE_ALL, .proc/close_all)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_RETURN_INVENTORY, .proc/signal_return_inv)
+	RegisterSignal(target, COMSIG_CLICK_ALT, .proc/on_alt_click)
+	RegisterSignal(target, COMSIG_MOUSEDROP_ONTO, .proc/mousedrop_onto)
+	RegisterSignal(target, COMSIG_MOUSEDROPPED_ONTO, .proc/mousedrop_receive)
+
+/datum/element/lazystorage/Detach(obj/item/storage/source, force)
+	. = ..()
+	UnregisterSignal(source, COMSIG_CONTAINS_STORAGE)
+	UnregisterSignal(source, COMSIG_IS_STORAGE_LOCKED)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_SHOW)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_INSERT)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_CAN_INSERT)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_TAKE_TYPE)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_FILL_TYPE)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_SET_LOCKSTATE)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_TAKE)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_QUICK_EMPTY)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_HIDE_FROM)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_HIDE_ALL)
+	UnregisterSignal(source, COMSIG_TRY_STORAGE_RETURN_INVENTORY)
+	UnregisterSignal(source, COMSIG_CLICK_ALT)
+	UnregisterSignal(source, COMSIG_MOUSEDROP_ONTO)
+	UnregisterSignal(source, COMSIG_MOUSEDROPPED_ONTO)
```

Comment:
Just do one single big UnresgisterSignal call with a list containing all the comsigs as second arg.

Reply:
I'll do it for the final commit for Unregister, I kind of want to keep the lines like this to better keep track of everything, I will probably remove/add some signals yet

Reply:
@vuonojenmustaturska 
```
UnregisterSignal(source, list(
	COMSIG_TRY_STORAGE_CAN_INSERT,
	COMSIG_TRY_STORAGE_QUICK_EMPTY,
))
```